core/src/main/java/com/linecorp/armeria/client/RedirectConfigBuilder.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.function.BiPredicate;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A builder for {@link RedirectConfig}.
+ */
+@UnstableApi
+public final class RedirectConfigBuilder {
+
+    private static final BiPredicate<ClientRequestContext, String> allowAllDomains = (ctx, domain) -> true;
+    static final BiPredicate<ClientRequestContext, String> allowSameDomain = (ctx, domain) -> {
+        final Endpoint endpoint = ctx.endpoint();
+        if (endpoint == null) {
+            return false;
+        }
+        return endpoint.host().contains(domain);
+    };
+
+    private RedirectRule redirectRule = RedirectRule.of();
+    private int maxRedirects = Integer.MAX_VALUE;
```

Comment:
How about setting 5 as the default value which is specified in https://datatracker.ietf.org/doc/html/rfc2068#section-10.3